### PR TITLE
Handle replication cases where the remote store metadata file is not present

### PR DESCRIPTION
**Pull request by ankitkala**
### Description
Handle the case in segrep with remote where primary hasn't refreshed yet. This is required because we trigger a force sync [here](https://github.com/opensearch-project/OpenSearch/blob/8eea7b986453271cd227a7b98ee09e70d5e28634/server/src/main/java/org/opensearch/indices/recovery/RecoverySourceHandler.java#L838)

### Related Issues
Resolves https://github.com/opensearch-project/OpenSearch/issues/8325

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -101,6 +101,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Support transport action names when registering NamedRoutes ([#7957](https://github.com/opensearch-project/OpenSearch/pull/7957))
 - Create concept of persistent ThreadContext headers that are unstashable ([#8291]()https://github.com/opensearch-project/OpenSearch/pull/8291)
 - Enable Partial Flat Object ([#7997](https://github.com/opensearch-project/OpenSearch/pull/7997))
+- Handle replication cases where the remote store metadata file is not present ([#8433](https://github.com/opensearch-project/OpenSearch/pull/8433))
```
Comment (gbbafna) [inline CHANGELOG.md:104]: we should skip change log for non user facing changes .

File: server/src/main/java/org/opensearch/indices/replication/RemoteStoreReplicationSource.java [hunk 2]
```diff
@@ -53,8 +55,14 @@ public void getCheckpointMetadata(
         // TODO: Need to figure out a way to pass this information for segment metadata via remote store.
         final Version version = indexShard.getSegmentInfosSnapshot().get().getCommitLuceneVersion();
         try {
-            metadataMap = remoteDirectory.readLatestMetadataFile()
-                .getMetadata()
+            RemoteSegmentMetadata mdFile = remoteDirectory.readLatestMetadataFile();
+            // During initial recovery flow, the remote store might not have metadata as primary hasn't uploaded anything yet.
+            if (mdFile == null && indexShard.state().equals(IndexShardState.STARTED) == false) {
```
Comment (mch2) [inline server/src/main/java/org/opensearch/indices/replication/RemoteStoreReplicationSource.java:60]: what about relocation case? If md is null we would want to bail here or break relocation elsewhere?

Comment (ankitkala) [reply server/src/main/java/org/opensearch/indices/replication/RemoteStoreReplicationSource.java:60]: Added tracking issue for follow up: https://github.com/opensearch-project/OpenSearch/issues/8681

Comment (mch2) [reply server/src/main/java/org/opensearch/indices/replication/RemoteStoreReplicationSource.java:60]: I don't think my concern here makes sense, we would not be going through the peer recovery if the shard were started so we are ok.

**Review by mch2 — APPROVED** [review]

**Comment by mch2** [comment]
Not sure how check was passing this test passed the checks here - i'd expect not having the segrep flag enabled to fail this test 100% of the time.  Anyway - https://github.com/opensearch-project/OpenSearch/pull/8704 should fix this.

**Pull request merged**